### PR TITLE
Add firestore

### PR DESCRIPTION
**Pull request by jaegerpicker**
Generated a new client from swagger for Firestore.

### CODE REVIEW

**Comment by jaegerpicker** [comment]
Veebie, my employer has signed the the CLA.

**Comment by jaegerpicker** [comment]
It's signed!

File: clients/firestore/lib/google_api/firestore/v1beta1/model/batch_get_documents_request.ex [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  - documents ([String.t]): The names of the documents to retrieve. In the format: &#x60;projects/{project_id}/databases/{database_id}/documents/{document_path}&#x60;. The request will fail if any of the document is not a child resource of the given &#x60;database&#x60;. Duplicate names will be elided. Defaults to: `null`.
+  - mask (DocumentMask): The fields to return. If not set, returns all fields.  If a document has a field that is not present in this mask, that field will not be returned in the response. Defaults to: `null`.
+  - newTransaction (TransactionOptions): Starts a new transaction and reads the documents. Defaults to a read-only transaction. The new transaction ID will be returned as the first response in the stream. Defaults to: `null`.
+  - readTime (String.t): Reads documents as they were at the given time. This may not be older than 60 seconds. Defaults to: `null`.
+  - transaction (binary()): Reads documents in a transaction. Defaults to: `null`.
+  """
+
+  defstruct [
+    :"documents",
```
Comment (chingor13) [inline clients/firestore/lib/google_api/firestore/v1beta1/model/batch_get_documents_request.ex]: It looks like this code hasn't gone through `mix format`. My guess is from file permissions on created files in the docker execution.

Can you run these files through `mix format`?

**Comment by carlodicelico** [comment]
@bshaffer Yes indeed! I'm using it right now, working beautifully! 🤓

File: clients/firestore/lib/google_api/firestore/v1beta1/connection.ex [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Copyright 2017 Google Inc.
```
Comment (chingor13) [inline clients/firestore/lib/google_api/firestore/v1beta1/connection.ex]: The year should be 2018. I realize it's hard-coded in the code generator and I'll need to come up with a global fix (it should be when the file is first created which will be hard to determine when we re-generate clients).

If you don't get to it, I'll fix it before a release.

File: clients/firestore/lib/google_api/firestore/v1beta1/model/batch_get_documents_request.ex (hunk 1 above)
Comment (carlodicelico) [reply clients/firestore/lib/google_api/firestore/v1beta1/model/batch_get_documents_request.ex]: No prob!

File: clients/firestore/lib/google_api/firestore/v1beta1/connection.ex (hunk 2 above)
Comment (carlodicelico) [reply clients/firestore/lib/google_api/firestore/v1beta1/connection.ex]: Fixing!

**Comment by carlodicelico** [comment]
@chingor13 Yes, we have some plans to work on getting a Firebase SDK using gRPC going when [grpc-elixir](https://github.com/tony612/grpc-elixir) is a bit more stable, based on the [Ruby](https://github.com/GoogleCloudPlatform/google-cloud-ruby/tree/master/google-cloud-firestore) version. In the interim, we don't really need Pub/Sub on our backend, we only need it on the clients and interacting with Firestore via REST is perfect for our specific use case at the moment.

I ran the formatter on the client and updated the copy right notices across all files in the project. Those changes are in two separate commits for obvious reasons 😆

**Comment by chingor13** [comment]
@carlodicelico Sorry, I meant to update the copyright year only on the new files (ones in the firebase client only). I opened #65 for me to deal with. If we change them here, it would have to be manually.

**Comment by carlodicelico** [comment]
Oh! I misinterpreted the phrase "on all the files". 😆 I'll fix that now.

**Pull request merged**